src/subdomains/core/buy-crypto/route/buy.controller.ts

```diff
@@ -109,16 +109,35 @@ export class BuyController {
 
   private async toPaymentInfoDto(userId: number, buy: Buy, dto: GetBuyPaymentInfoDto): Promise<BuyPaymentInfoDto> {
     const bankInfo = await this.getBankInfo(buy, dto);
+    const giroCode = this.generateGiroCode(buy, bankInfo, dto);
 
     return {
       ...bankInfo,
       sepaInstant: bankInfo.sepaInstant && buy.bankAccount.sctInst,
       remittanceInfo: buy.bankUsage,
       ...(await this.userService.getUserBuyFee(userId, buy.asset)),
       minDeposit: Config.transaction.minVolume.get(buy.asset, dto.currency.name),
+      giroCode,
     };
   }
 
+  private generateGiroCode(buy: Buy, bankInfo: BankInfoDto, dto: GetBuyPaymentInfoDto): string | undefined {
+    if (Config.giroCode.disabledCountries.includes(buy.iban.slice(0, 2))) return undefined;
+    return `
+    ${Config.giroCode.service}
+    ${Config.giroCode.version}
+    ${Config.giroCode.encoding}
+    ${Config.giroCode.transfer}
+    ${bankInfo.bic}
+    DFX AG
+    ${bankInfo.iban}
+    ${dto.currency?.name}${dto.amount}
```

Comment:
Return `undefined`, if currency is not defined?

Reply:
The currency name could theoretically be undefined, as only the id is required in the DTO. It's set in `getBankInfo`, which is a bit too hidden IMO 🤔  maybe fetch it before line 111 and give it as parameter to `getBankInfo` and `generateGiroCode`?